contrib/babelfishpg_tsql/src/pl_handler.c

```
@@ -3068,6 +3068,40 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 			{
 				ViewStmt *stmt = (ViewStmt *) parsetree;
 
+				/*
+				 * By default, TSQL view should be created with security_invoker
+				 * property. It adds security_invoker option to the options list.
+				 */
+				if (sql_dialect == SQL_DIALECT_TSQL && IS_TDS_CLIENT() && (stmt->createOrAlter || stmt->replace == false))
+				{
+					ListCell   *lc;
+					bool        security_invoker_found = false;
+
+					/* Check if security_invoker option already exists */
+					foreach(lc, stmt->options)
+					{
+						DefElem    *defel = (DefElem *) lfirst(lc);
+
+						if (strcmp(defel->defname, "security_invoker") == 0)
```

Can we ever get this security_invoker option for TSQL created views?

---

No

---

@pranavJ23 @HarshLunagariya.
Will dump restore in MVU change existing views to invoker ? Also should we throw an error if we find security definer option here since that is an unexpected scenario ?

---

No, Keeping it as it is since we decided not to change the configuration of existing views.

---

Just an observation: since we are removing the check ```stmt->replace```, as per the suggestion above: https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/4015/files#r2325291807 
If we do create or alter view then the existing views will also change to invoker

---

Even with that check, CREATE OR ALTER would have changed existing views to invoker right ?

---

Right, alter view would have definer(with check stmt->replace) instead of definer (not create or alter, my fault)
the testfile output [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/4015/files#diff-73874d5e73d54d043be4ce48051ce2786e0dd87638f4500767d5fd26eca42d53R48)()
was definer with check !stmt->replace but now it's invoker, which makes sense